src/main/scala/com/mozilla/spark/sql/hyperloglog/aggregates/HyperLogLogMerge.scala

```diff
@@ -56,7 +71,34 @@ class HyperLogLogMerge extends UserDefinedAggregateFunction {
   }
 
   def evaluate(buffer: Row): Any = {
-    val state = new DenseHLL(buffer.getAs[Int](1), new Bytes(buffer.getAs[Array[Byte]](0)))
+    val state = buffer(0) match {
+      case null => emptyHll
+      case o => new DenseHLL(buffer.getAs[Int](1), new Bytes(buffer.getAs[Array[Byte]](0)))
+    }
     com.twitter.algebird.HyperLogLog.toBytes(state)
   }
 }
+
+/**
+ * This class adds the capability to take in
+ * another Boolean column and only adds the
+ * associated hll if the Boolean column is `True`.
+ */
+class FilteredHyperLogLogMerge extends HyperLogLogMerge {
+
+  override def inputSchema: org.apache.spark.sql.types.StructType =
+    StructType(StructField("value", BinaryType) :: StructField("filtered", BooleanType) :: Nil)
+
+  override def update(buffer: MutableAggregationBuffer, input: Row): Unit = {
+    if (input(0) != null && input(1) != null && input.getAs[Boolean](1)) {
+      val hll = HyperLogLog.fromBytes(input.getAs[Array[Byte]](0)).toDenseHLL
+
+      if (buffer(0) != null) {
```

Comment:
This block (`if (buffer(0 != null)...`) is repeated in the parent`HyperLogLogMerge` update function as well - can we extract the common chunk to a separate function?